docs/tutorials/control/air-quality-fleet.md

```diff
@@ -20,766 +20,718 @@ cost: 200
+// Instantiate data_client and get all
+// data tagged with "air-quality" from your location
+const client = await VIAM.createViamClient(opts);
+const machine = await client.appClient.getRobot(machineId);
+const locationID = machine?.location;
+const orgID = (await client.appClient.listOrganizations())[0].id;
+
+const myDataClient = client.dataClient;
+const query = {
+  $match: {
+    tags: "air-quality",
+    location_id: locationID,
+    organization_id: orgID,
+  },
+};
+const match = { $group: { _id: "$robot_id" } };
+// Get a list of all the IDs of machines that have collected air quality data
+const BSONQueryForMachineIDList = [
+  BSON.serialize(query),
+  BSON.serialize(match),
+];
+let machineIDs: any = await myDataClient?.tabularDataByMQL(
+  orgID,
+  BSONQueryForMachineIDList,
+);
+// Get all the air quality data
+const BSONQueryForData = [BSON.serialize(query)];
+let measurements: any = await myDataClient?.tabularDataByMQL(
+  orgID,
+  BSONQueryForData,
+);
+```
+
+{{% /tablestep %}}
+{{% tablestep number=2 %}}
+
+For this project, your dashboard will display the average of the last five readings from each air sensor.
+You need a function to calculate that average.
+The data returned by the query is not necessarily returned in order, so this function must put the data in order based on timestamps before averaging the last five readings.
+
+Paste the following code into <file>main.ts</file> after the end of your main function, in place of `// <Insert getLastFewAv function definition here in later steps>`:
+
+```typescript {class="line-numbers linkable-line-numbers"}
+// Get the average of the last five readings from a given sensor
+async function getLastFewAv(all_measurements: any[], machineID: string) {
+  // Get just the data from this machine
+  let measurements = new Array();
+  for (const entry of all_measurements) {
+    if (entry.robot_id == machineID) {
+      measurements.push({
+        PM25: entry.data.readings["pm_2.5"],
+        time: entry.time_received,
+      });
+    }
+  }
+
+  // Sort the air quality data from this machine
+  // by timestamp
+  measurements = measurements.sort(function (a, b) {
+    let x = a.time.toString();
+    let y = b.time.toString();
+    if (x < y) {
+      return -1;
+    }
+    if (x > y) {
+      return 1;
+    }
+    return 0;
+  });
+
+  // Add up the last 5 readings collected.
+  // If there are fewer than 5 readings, add all of them.
+  let x = 5; // The number of readings to average over
+  if (x > measurements.length) {
+    x = measurements.length;
+  }
+  let total = 0;
+  for (let i = 1; i <= x; i++) {
+    const reading: number = measurements[measurements.length - i].PM25;
+    total += reading;
+  }
+  // Return the average of the last few readings
+  return total / x;
+}
+```
+
+{{% /tablestep %}}
+{{% tablestep number=3 %}}
+
+Now that you've defined the function to sort and average the data for each machine, you're done with all the `dataClient` code.
```

Comment:
Should a bunch of the instances of "each machine" here and later, including in code comments, actually be "the machine" since the app only connects to one machine at a time?

Reply:
going to keep as is because the code still does that and if you read it as each machine you deploy individually i think it's ok. Hopefully soon there'll be acess to multiple machines anyway